src/core/grid_based_algorithms/lb_interface.hpp

```diff
@@ -58,14 +54,19 @@ void lb_lbfluid_set_rng_state(uint64_t counter);
 const Lattice &lb_lbfluid_get_lattice();
 
 /**
- * @brief Get the global variable lattice_switch which defines wether NONE, CPU
- * or GPU LB is active.
+ * @brief Get the global variable @ref lattice_switch.
+ *
+ * Function to be called from Cython.
+ */
+int lb_lbfluid_get_lattice_switch_cython();
```

Comment:
Why is this function needed? scoped enums can be used in cython, please deal with cython stuff in the cython code.

Reply:
I already looked for a better alternative last week, but Cython support for C-style enums is limited:
* `cdef enum ActiveLB: NONE = 0, CPU, GPU` in .pxd, then in .pyx:
   * `fun(ActiveLB.CPU)` -> `error: ‘CPU’ was not declared in this scope: fun(CPU)`
   * `fun(<ActiveLB> ActiveLB.CPU)` -> `error: ‘CPU’ was not declared in this scope: fun((enum ActiveLB)CPU)`
* `cdef enum enum_ActiveLB "ActiveLB": NONE = 0, CPU, GPU` in .pxd, then in .pyx:
   * `fun(enum_ActiveLB.CPU)` -> `error: ‘CPU’ was not declared in this scope: fun(CPU)`
* same observations with `ctypedef` instead of `cdef`
* there is a namespace for pure Python enums, though

Is the new solution with two enums (f009b72) satisfactory?

Reply:
You can use them like this (example from the LB interface):

```cython
        cdef cppclass InterpolationOrder:
            pass

    cdef extern from "grid_based_algorithms/lb_interpolation.hpp" namespace "InterpolationOrder":
        cdef InterpolationOrder linear
        cdef InterpolationOrder quadratic
```

I think this should also work for return values.

Reply:
Ok, it should work now. It's kind of annoying the exported enum values cannot be grouped in a Python namespace (`cdef ActiveLB ActiveLB_CPU "CPU"` didn't work).

Reply:
Yes, it's not nice. Did you see https://github.com/cython/cython/issues/1603#issuecomment-324602083 ?

Reply:
Saw it last week, but forgot to try it... Let's see if all compilers accept it.

Reply:
Looks like a no on Cython 0.23.4 (excepted container `empty`), but a yes on 0.25.2 and above. Should I just revert the last commit?

Reply:
Yes, just revert to the solution without the namespace.